### PR TITLE
Fix/var tests

### PR DESCRIPTION
**Pull request by willmclaren**

### CODE REVIEW

File: lib/EnsEMBL/REST/Model/Variation.pm [hunk 1]
```diff
@@ -192,7 +192,7 @@ sub get_gene_phenotype_info {
 
   foreach my $phen (@pfs) {
     my $hash = $self->phen_as_hash($phen);
-    my $key = join("", sort values %$hash);
+    my $key = join("", sort grep {$_} values %$hash);
```
Comment (nerdstrike) [inline lib/EnsEMBL/REST/Model/Variation.pm]: This will need a comment, or someone will take it out in future since it appears to do nothing.

Comment (willmclaren) [inline lib/EnsEMBL/REST/Model/Variation.pm]: Why would you say it appears to do nothing?

Comment (nerdstrike) [inline lib/EnsEMBL/REST/Model/Variation.pm]: I am putting myself of future coder who goes "huh?".

Comment (willmclaren) [inline lib/EnsEMBL/REST/Model/Variation.pm]: grep {$_} means your checking it has a non-0 and non-null value. How would you assume that does nothing unless you don't know what grep does?

Comment (nerdstrike) [inline lib/EnsEMBL/REST/Model/Variation.pm]: If you understand all Perl primitives, you understand all code right? No comments about circumstance required at all QED. Commenting on at least the purpose of the sub{} will save someone some brain glucose in future.

**Pull request merged**